syntax = "proto3";

service ClientService {
  rpc ListFiles(Empty) returns (FileList) {}
  rpc CreateFile(FileData) returns (Status) {}
  rpc Open(FilePath) returns (Status) {}
  rpc Close(FilePath) returns (Status) {}
  rpc Read(FilePath) returns (FileData) {}
  rpc Write(FileData) returns (Status) {}
  rpc FilePartition(FilePath) returns (PartitionedFile) {}
  rpc JoinPartitionedFiles(PartitionedFile) returns (FileData) {}
}

message Empty {}

message FileList {
  repeated string files = 1;
}

message FilePath {
  string path = 1;
}

message FileData {
  string name = 1;
  bytes data = 2;
}

message PartitionedFile {
  string name = 1;
  repeated bytes partitions = 2;
}

message Status {
  bool success = 1;
  string message = 2;
}
